### PR TITLE
DIV-4052: Remove non numerical characters from case ref number when linking case

### PR DESCRIPTION
**Pull request by johnoneill46**
Regex to remove non numerical characters from case ref number when linking case.

Fixes prod issue where users are unable to correctly link their cases when entering hyphens etc

### CODE REVIEW

File: steps/capture-case-and-pin/CaptureCaseAndPin.step.js [hunk 1]
```diff
@@ -27,11 +27,13 @@ class CaptureCaseAndPin extends Question {
...
         errors.referenceNumberRequired,
         Joi.string()
+          .replace(/\D/gi, '')
           .length(referenceNumberLength)
           .required())
       .joi(
         errors.referenceNumberDigitsOnly,
         Joi.string()
+          .replace(/\D/gi, '')
           .regex(/^[0-9]+$/)
```
Comment (hosainnet) [inline steps/capture-case-and-pin/CaptureCaseAndPin.step.js]: I don't think this regex/the whole 2nd `.joi` is relevant anymore, given that we're replacing any non-digit in the first joi. Subsuqently, we can remove `referenceNumberDigitsOnly` and any references to it as well

Comment (hosainnet) [reply steps/capture-case-and-pin/CaptureCaseAndPin.step.js]: We might also have to update this scenario https://github.com/hmcts/div-respondent-frontend/blob/master/test/e2e/paths/captureCaseAndPin.js#L17
(I know.. we should be running those tests in PRs but that's a separate story...)

Comment (johnoneill46) [reply steps/capture-case-and-pin/CaptureCaseAndPin.step.js]: In the CaptureCaseAndPin.js we're not actually storing/updating the value of the string here, we are just removing the redundant characters so it passes the validation. The actual logic of updating the case reference number is done in caseOrchestration.js

Comment (hosainnet) [reply steps/capture-case-and-pin/CaptureCaseAndPin.step.js]: Cleared up after IRL conversation 👍

**Pull request merged**